app/javascript/components/App.js

```diff
@@ -1,60 +1,71 @@
-import React, { Component } from 'react'
-import {
-  BrowserRouter as Router,
-  Route,
-  Switch
-} from 'react-router-dom'
-import AboutUs from './pages/AboutUs'
-import Footer from './components/Footer'
-import Header from './components/Header'
-import Home from './pages/Home'
-import LocationEdit from './pages/locations/LocationEdit'
-import LocationIndex from './pages/locations/LocationIndex'
-import LocationNew from './pages/locations/LocationNew'
-import LocationShow from './pages/locations/LocationShow'
-import NotFound from './pages/NotFound'
+import React, { Component } from "react";
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
+import AboutUs from "./pages/AboutUs";
+import Footer from "./Components/Footer";
+import Header from "./Components/Header";
+import Home from "./pages/Home";
+import LocationEdit from "./pages/Locations/LocationEdit";
+import LocationIndex from "./pages/Locations/LocationIndex";
+import LocationNew from "./pages/Locations/LocationNew";
+import LocationShow from "./pages/Locations/LocationShow";
+import NotFound from "./pages/NotFound";
+import UserLocation from "./pages/Locations/UserLocation";
 
 class App extends Component {
   constructor(props) {
-    super(props)
+    super(props);
     this.state = {
-      locations: []
-    }
+      locations: [],
+    };
   }
 
-  componentDidMount () {
-    this.readlocation()
+  componentDidMount() {
+    this.readlocation();
   }
 
-  readlocation= () => {
-    fetch ("/locations")
-    .then(response => response.json())
-    .then(payload => this.setState({locations: payload}))
-    .catch(errors => console.log('locations read error: ', errors))
-  }
+  readlocation = () => {
+    fetch("/locations")
+      .then((response) => response.json())
+      .then((payload) => this.setState({ locations: payload }))
+      .catch((errors) => console.log("locations read error: ", errors));
+  };
 
   render() {
     const {
       logged_in,
       current_user,
       new_user_route,
       sign_in_route,
-      sign_out_route
-    } = this.props
+      sign_out_route,
+    } = this.props;
+    console.log("logged_in:", logged_in);
+    console.log("current_user:", current_user);
```

Comment:
Please remove all console logs from your code. I am also not sure if you need to destructure all of your routes. It seems like you are only using current_user and logged_in.